llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -647,6 +647,38 @@ void setSyclFixedTargetsMD(const std::vector<Function *> &EntryPoints,
     F->setMetadata("sycl_fixed_targets", MDN);
 }
 
+void collectVirtualFunctionSetInfo(
+    Function &F, StringMap<std::vector<Function *>> &VirtualFunctionSets) {
```

Comment:
If it won't require refactoring of existing code (due to shared data structures, or functions), I would add `const` to `F` argument (and everywhere else where it is applicable, i.e. to `processDeclaredVirtualFunctionSets` arguments as well)